### PR TITLE
[Feature][Connector-V2][Lemlist]Add Lemlist source connector

### PR DESCRIPTION
**Pull request by TaoZex**
## Purpose of this pull request

1.Create a common authentication method for SaaS,ther authentication methods will continue to be added later.

2.Add Lemlist source connector

config :
2da4ed17ac6e832fe432227dffc23ab

Without using fields:
686fc631cbec57a870d5cf2717fbca1

using fields:
33b492c185edd1fc6a3959448ac496e

## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/incubator-seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/incubator-seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/incubator-seatunnel/blob/dev/seatunnel-dist/pom.xml)

### CODE REVIEW

**Comment by hailin0** [comment]
Thanks, can you help add the e2e test

reference
https://github.com/apache/incubator-seatunnel/pull/3340

**Review by ashulin — COMMENTED** [review]
You need to add `Lemlist` source to `plugin-mapping.properties`

**Comment by TaoZex** [comment]
> You need to add `Lemlist` source to `plugin-mapping.properties`

Thank you so much for your help.

File: seatunnel-connectors-v2/connector-http/connector-http-lemlist/src/main/java/org/apache/seatunnel/connectors/seatunnel/lemlist/source/LemlistSourceFactory.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    }
+
+    @Override
+    public OptionRule optionRule() {
+        return OptionRule.builder()
+                .required(LemlistSourceConfig.URL)
+                .required(LemlistSourceConfig.PASSWORD)
+                .optional(LemlistSourceConfig.RETRY)
+                .optional(LemlistSourceConfig.RETRY_BACKOFF_MAX_MS)
+                .optional(LemlistSourceConfig.RETRY_BACKOFF_MULTIPLIER_MS)
```
Comment (EricJoy2048) [inline seatunnel-connectors-v2/connector-http/connector-http-lemlist/src/main/java/org/apache/seatunnel/connectors/seatunnel/lemlist/source/LemlistSourceFactory.java:52]: Why the option number not equals to the document?

**Comment by EricJoy2048** [comment]
Please add e2e test reference https://github.com/apache/incubator-seatunnel/pull/3340

**Comment by TaoZex** [comment]
> Please add e2e test reference #3340

Thanks. I have added e2e test.

**Comment by TaoZex** [comment]
> Thanks, can you help add the e2e test
> 
> reference #3340

Thanks. I have added e2e test.PTAL.

File: seatunnel-e2e/seatunnel-connector-v2-e2e/connector-http-e2e/src/test/resources/mockserver-lemlist-config.json [hunk 2]
```diff
@@ -0,0 +1,109 @@
+// https://www.mock-server.com/mock_server/getting_started.html#request_matchers
+
+[
+  {
+    "httpRequest": {
+      "method" : "GET",
+      "path": "/example/lemlist/team"
```
Comment (hailin0) [inline seatunnel-e2e/seatunnel-connector-v2-e2e/connector-http-e2e/src/test/resources/mockserver-lemlist-config.json]: The path should be the same as lemlist

File: seatunnel-e2e/seatunnel-connector-v2-e2e/connector-http-e2e/src/test/resources/mockserver-lemlist-config.json [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+            "plan": "eat food"
+          }
+        }
+      ]
+    }
+  },
+  {
+    "httpRequest": {
+      "method" : "GET",
+      "path": "/example/lemlist/campaigns"
```
Comment (hailin0) [inline seatunnel-e2e/seatunnel-connector-v2-e2e/connector-http-e2e/src/test/resources/mockserver-lemlist-config.json]: The path should be the same as lemlist

Comment (TaoZex) [reply seatunnel-e2e/seatunnel-connector-v2-e2e/connector-http-e2e/src/test/resources/mockserver-lemlist-config.json]: done

**Review by hailin0 — APPROVED** [review]

**Review by TyrantLucifer — APPROVED** [review]

**Pull request merged**